app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.java

```diff
@@ -37,8 +38,10 @@
     private static final String TRACKS_NODE = "data/tracks";
     private static final String VENUE_INFO_NODE = "data/venue";
     private static final String TRACKS_BY_ID_NODE = "data/tracks/%1$s";
-    private static final String FAVORITES_NODE = "user/%1$s/";
-    private static final String FAVORITES_BY_ID_NODE = "user/%1$s/favorites/%2$s";
+    private static final String FAVORITES_NODE = "user/%1$s/favorites";
+    private static final String FAVORITES_BY_ID_NODE = "user/%1$s/favorites/map/%2$s";
```

Comment:
Oh ok I see. Well maybe the easy solution is to call the maps something more descriptive, but tbh I would rather get rid of one level and have them directly in each user's node:

```javascript
{
  "data": { /* blah blah */ },
  "user": {
    "any_user_id": {
      "favorites": {
        "event_id_1": true,
        "event_id_4": true,
        "event_id_56": true,
        "event_id_98": true
      },
      "achievements": {
        "achievement_id_1": 12364546798,
        "achievement_id_2": 12364745700
      }
    }
  }
}
```

The paths would then be, if I am not mistaken:

```java
    private static final String FAVORITES_NODE = "user/%1$s/favorites";
    private static final String FAVORITES_BY_ID_NODE = "user/%1$s/favorites/%2$s";
    private static final String ACHIEVEMENTS_NODE = "user/%1$s/achievements";
    private static final String ACHIEVEMENTS_BY_ID_NODE = "user/%1$s/achievements/%2$s";
```

and `FirebaseFavorites` would become `FirebaseUser`:

```java
public class FirebaseUserData {

    public static FirebaseUserData empty() {
        return new FirebaseUserData(Collections.emptyMap(), Collections.emptyMap());
    }

    public FirebaseUserData() {
    }

    public FirebaseUserData(Map<String, Boolean> favorites, Map<String, Boolean> achievements) {
        this.favorites = favorites;
        this.achievements = achievements;
    }

    public Map<String, Boolean> favorites;

    public Map<String, Boolean> achievements;

    // ...
}
```

Could this work?

CC @fourlastor